include/mfmg/dealii/lanczos.templates.hpp

```diff
@@ -351,15 +351,12 @@ template <typename OperatorType, typename VectorType>
 void Lanczos<OperatorType, VectorType>::details_set_initial_guess(
     VectorType &initial_guess, int seed)
 {
-  // Set initial guess to constant vector to try to capture "smooth" eigenmodes
-  // of PDE problems
-  initial_guess = 1.;
-
   // Add random noise to the guess
   std::mt19937 gen(seed);
   std::uniform_real_distribution<double> dist(0, 1);
   std::transform(initial_guess.begin(), initial_guess.end(),
-                 initial_guess.begin(), [&](auto &v) { return v + dist(gen); });
+                 initial_guess.begin(),
+                 [&](auto &v) { return (1. + dist(gen)) * v; });
```

Comment:
Need an extensive comment here about the care that went into this very fragile line. Specifically, it hides the idea that it preserves the zeros in the initial guess that are already there, which correspond to constraints.